src/controller/SearchMovieListController.ts

```diff
@@ -0,0 +1,106 @@
+import { getSearchMovieResult } from "../api/getSearchMovieResult";
+import MovieEmptySection from "../component/MovieEmptySection";
+import MovieItem from "../component/MovieItem";
+import MovieListSection from "../component/MovieListSection";
+import SkeletonMovieItem from "../component/Skeleton/SkeletonMovieItem";
+import SkeletonMovieListSection from "../component/Skeleton/SkeletonMovieListSection";
+import { IMovieItem, IMovieResult } from "../types/movieResultType";
+
+class SearchMovieListController {
+  mainElement;
+  searchText;
+  page = 0;
+
+  constructor(mainElement: HTMLElement, searchText: string) {
+    this.mainElement = mainElement;
+    this.searchText = searchText;
+
+    this.render();
+  }
+
+  bindEvents() {
+    const seeMoreElement = this.mainElement.querySelector(".see-more");
+    seeMoreElement?.addEventListener("click", () => {
+      this.addMovieList();
+    });
+  }
+
+  async fetchMovies() {
+    const {
+      page: newPage,
+      total_pages: totalPage,
+      results: movieList,
+    }: IMovieResult = await getSearchMovieResult(
+      this.searchText,
+      this.page + 1,
+    );
+    this.page = newPage;
+
+    const hasMore = newPage !== totalPage;
+
+    return { movieList, hasMore };
+  }
+
+  async render() {
+    this.renderSkeleton();
+
+    const { movieList, hasMore } = await this.fetchMovies();
+
+    this.renderSearchMovieList({ movieList, hasMore });
+
+    this.bindEvents();
+  }
+
+  renderSkeleton() {
+    const skeletonSectionElement = SkeletonMovieListSection();
+    this.mainElement.replaceChildren(skeletonSectionElement);
+  }
+
+  renderSearchMovieList({
+    movieList,
+    hasMore,
+  }: {
+    movieList: IMovieItem[];
+    hasMore: boolean;
+  }) {
+    let sectionElement;
+    if (movieList.length !== 0) {
+      sectionElement = MovieListSection({
+        title: `"${this.searchText}" 검색 결과`,
+        movieList,
+        hasMore,
+      });
+    } else {
+      sectionElement = MovieEmptySection(`"${this.searchText}" 검색 결과`);
+    }
+
+    this.mainElement.replaceChildren(sectionElement);
+  }
+
+  async addMovieList() {
+    const movieListContainer = this.mainElement.querySelector("ul");
+    if (!movieListContainer) return;
+
+    // 스켈레톤 추가
+    const skeletonElements = Array.from({ length: 20 }, () =>
+      SkeletonMovieItem(),
+    );
+    skeletonElements.forEach((skeletonElement) => {
+      movieListContainer?.appendChild(skeletonElement);
+    });
+
+    const { movieList, hasMore } = await this.fetchMovies();
+
+    // 스켈레톤 제거 후 새로운 영화 추가
+    skeletonElements.forEach((skeletonElement) => {
+      skeletonElement.remove();
+    });
+    movieList.forEach((movie) => {
+      movieListContainer?.appendChild(MovieItem(movie));
+    });
```

Comment:
스켈레톤과 MovieItem 을 하나하나 집어넣고 빼기보다는 다음과 같은 방식으로 더 단순화 할 수 있을 것 같습니다.

```suggestion
    // 스켈레톤 추가
    const skeletonElements = Array.from({ length: 20 }, () =>
      SkeletonMovieItem(),
    );

    movieListContainer.append(...skeletonElements);

    const { movieList, hasMore } = await this.fetchMovies();

    movieListContainer.replaceChildren(
      ...movieList.map((movie) => MovieItem(movie)),
    );
```

Reply:
appendChild 말고 append를 통해서 여러 개를 한꺼번에 추가하는 방법을 새롭게 알게되었습니다 감사합니다 ㅎㅎ

```
movieListContainer.replaceChildren(
  ...movieList.map((movie) => MovieItem(movie)),
);
```

하지만 위 코드에서 replaceChildren을 해버리면 추가되는 movieList 말고 기존 list 역시 replace 대상이 되기 때문에, 여기는 remove -> append 해주는 방식을 유지하였습니다! 그리고 skeleton의 개수와 추가되는 movieList의 개수가 다를 수 있기 때문에, 또한 해당 방법을 사용하지 사용하지 못하였습니다 ㅜㅜ

Reply:
아하 제가 코드를 제대로 파악하지 못했었나보군요. 직접 수정 후 로컬에서 돌려봤을 때는 문제 없는 것처럼 보였는데 놓친 부분이 있었나봅니다.